src/core/js/views/questionView.js

```diff
@@ -279,6 +282,38 @@ define([
         // be a full reset
         resetQuestion: function() {},
 
+        determineButtonState: function() {
+            if (this.model.get('_isInteractionComplete')) {
+                if (this.model.get('_isCorrect')) {
+                    this.model.set('_buttonState', 'correct');
+                } else if (!this.model.get('_canShowModelAnswer')) {
+                    this.model.set('_buttonState', 'incorrect');
+                } else {
+                    this.model.set('_buttonState', 'showCorrectAnswer');
+                }
+            } else {
+                if (this.model.get('_isEnabled')) {
+                    this.model.set('_buttonState', 'submit');
+                } else {
+                    this.model.set('_buttonState', 'reset');
+                }
+            }
+        },
+
+        refresh: function() {
+            this.renderState();
+              
+            this.determineButtonState();
```

Comment:
with:

```javascript
this.model.set("_buttonState", this.model.getButtonState());
```

instead of:
```javascript
this.determineButtonState();
```